app/src/main/java/com/mithridat/nonoconverter/backend/nonogram/Nonogram.java

```diff
@@ -0,0 +1,332 @@
+     * with column number i and row number j
+     *
+     * @param i - column number
+     * @param j - row number
+     * @return top nonogram grid cell with column number i and row number j
+     */
+    public int getTopColValue(int i, int j) {
+        return _top[i][j];
+    }
+
+    /**
+     * Method for getting _left[i][j] or _top[i][j]
+     *
+     * @param i - i
+     * @param j - j
+     * @param type - ROW, if _left
+     *               COL, if _top
+     * @return _left[i][j] or _top[i][j]
+     */
+    public int getValue(int i, int j, int type) {
+        if (type == ROW) return getLeftRowValue(i, j);
+        return getTopColValue(i, j);
+    }
+
+    /**
+     * Method for checking correctness of filling of the field
+     *
+     * @return true, if filling of the field is correct
+     *         false, otherwise
+     */
+    public boolean checkCorrectness() {
+        int rows = _field.getRows(), cols = _field.getCols();
+        Field field = _field.copy();
+        for (int i = 0; i < rows; i++) {
+            for (int j = 0; j < cols; j++) {
+                if (field.getColor(i, j) == EMPTY) {
+                    field.setColor(i, j, 0);
+                }
+            }
+        }
+        return checkNonogramGrid(field, _left, ROW, rows, cols) &&
+                checkNonogramGrid(field, _top, COL, cols, rows);
+    }
+
+    /**
+     * Method for getting field of the puzzle
+     *
+     * @return field of the puzzle
+     */
+    public Field getField() {
+        return _field;
+    }
+
+    /**
+     * Method for clearing field of the puzzle
+     */
+    public void clearField() {
+        _field.clear();
+    }
+
+    /**
+     * Method for getting new nonogram that contains field with colors in cells
+     *
+     * @return new nonogram that contains field with colors in cells
+     */
+    public Nonogram translateToColors() {
+        return new Nonogram(_field.translateToColors());
+    }
+
+    /**
+     * Method for initialising grid
+     */
+    private void initGrid() {
+        int w = _field.getCols(), h = _field.getRows();
+
+        _left = new int[h][];
+        _top = new int[w][];
+
+        fillNonogramGrid(_left, ROW, h, w);
+        fillNonogramGrid(_top, COL, w, h);
+    }
+
+    /**
+     * Method for filling one of the two grids from field
+     *
+     * @param grid - grid for filling
+     * @param type - ROW, if _left
+     *               COL, if _top
+     * @param outLim - limit for counter in outer loop
+     * @param inLim - limit for counter in inner loop
+     */
+    private void fillNonogramGrid(
+            int[][] grid,
+            int type,
+            int outLim,
+            int inLim) {
+        for (int i = 0; i < outLim; ++i) {
+            grid[i] = new int[0];
+            for (int j = 0, k, length = 0; j < inLim; ) {
+                k = _field.getAnotherColorIndex(i, j, 1, type);
+                if (_field.getColor(i, j, type) ==
```

Comment:
`==` is an operator. Check [Coding standard](https://github.com/nonoteam/NonoConverter/blob/master/Development.md#line-wrapping-strategies).